This fixes generation of GOTPCREL relocations, they had sometimes the
wrong offset.

This version should be safe even for non-ELF targets.
Po srp 26 21:36:33 CEST 2002  Jan Hubicka  <jh@suse.cz>
	* tc-i386.c (tc_i386_fix_adjustable): Recognize GOTPCREL properly.
Index: config/tc-i386.c
===================================================================
RCS file: /cvs/src/src/gas/config/tc-i386.c,v
retrieving revision 1.126
diff -c -3 -p -r1.126 tc-i386.c
*** gas/config/tc-i386.c	10 Aug 2002 14:49:48 -0000	1.126
--- config/tc-i386.c	26 Aug 2002 19:36:21 -0000
*************** tc_i386_fix_adjustable (fixP)
*** 1224,1229 ****
--- 1224,1235 ----
  	  && fixP->fx_pcrel))
      return 0;
  
+   /* The x86_64 GOTPCREL are represented as 32bit PCrel relocations
+      and changed later by validate_fix.  */
+   if (GOT_symbol && fixP->fx_subsy == GOT_symbol
+       && fixP->fx_r_type == BFD_RELOC_32_PCREL)
+     return 0;
+ 
    /* adjust_reloc_syms doesn't know about the GOT.  */
    if (fixP->fx_r_type == BFD_RELOC_386_GOTOFF
        || fixP->fx_r_type == BFD_RELOC_386_PLT32

